### PR TITLE
move to rabbot as base, instead of wascally

### LINKED ISSUE #25: Rabbot Support
Any plans on supporting the new library? https://www.npmjs.com/package/rabbot

### PR DESCRIPTION
**Pull request by mxriverlynn**
This pull request changes out the old, deprecated wascally library for the updated / maintained rabbot library.

and because of poor planning on my part, way back when i was doing this work, this PR also includes previous work on building out a new topology definition object. it should be backward compatible with previous versions of the rabbus API.

fixes #25

### CODE REVIEW

File: demos/connection.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 var util = require("util");
-var wascally = require("wascally");
+var rabbot = require("rabbot");
```
Comment (rhyslbw) [inline demos/connection.js:2]: Maybe call this variable `rabbit` to abstract the RabbitMQ driver

File: demos/send-receive/receive.js [hunk 2]
```diff
@@ -21,6 +21,11 @@ util.inherits(SomeReceiver, Rabbus.Receiver);
 // -----------------------------
 
 connection(function(){
+  receive();
+  receive();
```
Comment (rhyslbw) [inline demos/send-receive/receive.js]: Duplicate line?

File: UPGRADING.md [hunk 3]
```diff
@@ -3,6 +3,15 @@
...
 to / from various versions of Rabbus.
 
+### From v0.7.0 to v0.8.0
+
+With v0.8.0, Rabbus is now built on top of the [rabbot](https://github.com/arobson/rabbot) library,
+instead of the now deprecated wascally library.
+
+Update your projects to include the latest v1.x rabbot:
+
+`npm install rabbot --save`
```
Comment (rhyslbw) [inline UPGRADING.md:21]:
```
npm uninstall wascally --save
npm install rabbot --save
```

File: package.json [hunk 4]
```diff
@@ -33,22 +33,20 @@
...
   "dependencies": {
     "generic-middleware": "2.1.0",
-    "postal": "1.0.7",
+    "postal": "2.0.4",
     "underscore": "1.8.3",
-    "whistlepunk": "0.3.2"
-  },
-  "peerDependencies": {
-    "wascally": "*"
+    "whistlepunk": "0.3.3"
```
Comment (rhyslbw) [inline package.json:48]: See https://github.com/LeanKit-Labs/whistlepunk/issues/15#issuecomment-238353455

**Comment by rhyslbw** [comment]
Have you come across git-flow before? If not check this out: http://danielkummer.github.io/git-flow-cheatsheet/. In this case it would help to keep this PR free from [release](http://danielkummer.github.io/git-flow-cheatsheet/#release) concerns, and show you the value of treating the `develop` branch as a staging area for the next release. With this approach the version number stays the same in any feature or fix branches and changelog items added under a heading`vNext`.

Also considering node 4 is a dependency, have you considered using ES6 classes?

**Comment by mxriverlynn** [comment]
this branch is definitely broken

i didn't realize i had a bunch of unit tests disabled, when i made this PR. still have a lot of work to do, as the change from wascally to rabbot was definitely not just drop-in-done.

**Comment by mxriverlynn** [comment]
v0.8.0 is now updated to use rabbot instead of wascally

**Pull request merged**